okhttp/src/main/java/com/lightstep/tracer/shared/HttpCollectorClient.java

```diff
@@ -2,18 +2,99 @@
 
 import com.lightstep.tracer.grpc.ReportRequest;
 import com.lightstep.tracer.grpc.ReportResponse;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 
 class HttpCollectorClient extends CollectorClient {
-  HttpCollectorClient() {}
+    private static final MediaType protoMediaType = MediaType.parse("application/octet-stream");
+
+    private final AtomicReference<OkHttpClient> client;
+    private final AbstractTracer tracer;
+    private final URL collectorURL;
+    private final long deadlineMillis;
+
+    HttpCollectorClient(
+            AbstractTracer tracer,
+            URL collectorURL,
+            long deadlineMillis
+    ) {
+        this.client = new AtomicReference<>(start(deadlineMillis));
+        this.tracer = tracer;
+        this.collectorURL = collectorURL;
+        this.deadlineMillis = deadlineMillis;
+    }
+
+    @Override
+    ReportResponse report(ReportRequest request) {
+        try {
+            return fromResponse(this.client().newCall(toRequest(request)).execute());
+        } catch (Exception e) {
+            tracer.error("Exception sending report to collector: ", e);
+            return null;
+        }
+    }
+
+    @Override
+    void reconnect() {
```

Comment:
Not blocking, more of a note here: reconnect is hacky-ish way we came up with to do client-side load balancing for GRPC connections (resetting connections every 5 minutes theoretically round robins through all the backend servers). This is not needed for HTTP clients and seems wasteful, so we will need to come up with a better solution, maybe use reflection to figure out client impl and skip the reconnect [here](https://github.com/lightstep/lightstep-tracer-java-common/blob/master/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java#L219)

Reply:
Ya, we should probably make sure we round robin the connections anyway. HTTP2 will have persistent connections so we will have the same problem we had with GRPC. Although there might be a setting in okhttp to tell it how long you allow connections to live.